src/utils/DependencyUtils.ts

```diff
@@ -11,3 +29,62 @@ export const splitToNameAndVersion = (dependencyString: string): [string, string
     const fullName = parts.join('-');
     return [fullName, version];
 };
+
+export async function getFullDependencyList(
+    combos: ThunderstoreCombo[],
+    game: Game,
+    installedMods: ManifestV2[],
+    modPackDependencySetBuilderMode: ModPackDependencySetBuilderMode = ModPackDependencySetBuilderMode.USE_EXACT_VERSION
+) {
+    const originalCombos = [...combos];
+    for (const combo of originalCombos) {
+        const isModpack = combo.getMod().getCategories().some(value => value === "Modpacks");
+        const versionMode = isModpack ? modPackDependencySetBuilderMode : ModPackDependencySetBuilderMode.USE_LATEST_VERSION;
+
+        await buildDependencySet(combo.getVersion(), game, combos, versionMode);
+
+        // Only filter if this combo is not a modpack and there are some installed mods
+        if (!isModpack && installedMods.length > 0) {
+            const filteredCombos = combos.filter((dep) => 
+                !installedMods.some(installed => installed.getName() === dep.getMod().getFullName())
+            );
+            combos.length = 0;
+            combos.push(...filteredCombos);
```

Comment:
Wait, has this always been here? What's the idea of resetting `combos` to empty array on every iteration? We lose all the work done so far?

Reply:
This comment is unaddressed.

Reply:
Oops...
So yeah after scratching my head for hours and hours, I finally figured out why the filtering didn't work correctly.
Now the filtering logic is fixed and thus the `.length = 0` wasn't needed anymore.